### PR TITLE
Make statement tasks visible

### PR DESCRIPTION
**Pull request by EdwinKruglov**
## Changes in this PR

- fix a bug where tasks with a single statement step are not shown on the journey page
- move and rename `journeys/section_spec.rb` to `journeys/sections/order_spec.rb`
- add `journeys/sections/visible_tasks_spec.rb`

### CODE REVIEW

File: spec/features/journeys/sections/view_spec.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  let(:section) { create(:section, journey: journey) }
+
+  before do
+    user_is_signed_in(user: user)
+  end
+
+  context "when a task has one step" do
+    context "and it is a question" do
+      before do
+        single_question_task = create(:task, title: "Single question task", section: section)
```
Comment (peterdavidhamilton) [inline spec/features/journeys/sections/view_spec.rb:14]: Just use `task` for the variable name

File: app/views/journeys/show.html.erb [hunk 2]
```diff
@@ -12,28 +12,23 @@
...
-                  </span>
-                  <strong class="govuk-tag govuk-tag--grey app-task-list__tag" id="<%= task.step.status_id %>">
-                    <%= I18n.t("task_list.status.not_started") %>
-                  </strong>
-                <% end %>
-              </li>
-            <% end %>
-
+            <li class="app-task-list__item step__item" id="<%= task.step.id %>">
+              <% if task.step.answered? || task.step.acknowledged? %>
```
Comment (peterdavidhamilton) [inline app/views/journeys/show.html.erb:16]: Can we instead add a `completed?` predicate to the step that is type agnostic.

File: spec/features/journeys/sections/view_spec.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          expect(find("a.govuk-link")).to have_text "Single statement task"
+        end
+      end
+    end
+  end
+
+  context "when a task has multiple steps" do
+    context "and all are questions" do
+      before do
+        all_question_task = create(:task, title: "All question task", section: section)
```
Comment (peterdavidhamilton) [inline spec/features/journeys/sections/view_spec.rb:44]: Can we use "Task with only questions"

File: spec/features/journeys/sections/view_spec.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    context "and it is a question" do
+      before do
+        single_question_task = create(:task, title: "Single question task", section: section)
+        create(:step, :number, task: single_question_task)
+        visit "/journeys/#{journey.id}"
+      end
+
+      it "is visible" do
+        within "ul.app-task-list__items" do
+          expect(find("a.govuk-link")).to have_text "Single question task"
```
Comment (peterdavidhamilton) [inline spec/features/journeys/sections/view_spec.rb:21]: "Task with single question"

**Comment by peterdavidhamilton** [comment]
Would `sections/visible_tasks_spec.rb` be more applicable than `view_spec`

**Review by AndrewCowle — APPROVED** [review]

**Review by peterdavidhamilton — APPROVED** [review]

**Pull request merged**